pytest_fixtures/satellite_auth.py

```diff
@@ -73,24 +73,74 @@ def auth_source(module_org, module_loc, ldap_data):
 def auth_source_ipa(module_org, module_loc, ipa_data):
     return entities.AuthSourceLDAP(
         onthefly_register=True,
-        account=ipa_data['ldap_ipa_user_name'],
-        account_password=ipa_data['ldap_ipa_user_passwd'],
-        base_dn=ipa_data['ipa_base_dn'],
-        groups_base=ipa_data['ipa_group_base_dn'],
+        account=ipa_data['ldap_user_cn'],
+        account_password=ipa_data['ldap_user_passwd'],
+        base_dn=ipa_data['base_dn'],
+        groups_base=ipa_data['group_base_dn'],
         attr_firstname=LDAP_ATTR['firstname'],
         attr_lastname=LDAP_ATTR['surname'],
         attr_login=LDAP_ATTR['login'],
         server_type=LDAP_SERVER_TYPE['API']['ipa'],
         attr_mail=LDAP_ATTR['mail'],
         name=gen_string('alpha'),
-        host=ipa_data['ldap_ipa_hostname'],
+        host=ipa_data['ldap_hostname'],
         tls=False,
         port='389',
         organization=[module_org],
         location=[module_loc],
     ).create()
 
 
+@fixture
+def ldap_auth_source(request, module_org, module_loc, ad_data, ipa_data):
+    if request.param.lower() == 'ad':
+        # entity create with AD settings
+        entities.AuthSourceLDAP(
+            onthefly_register=True,
+            account=ad_data['ldap_user_name'],
+            account_password=ad_data['ldap_user_passwd'],
+            base_dn=ad_data['base_dn'],
+            groups_base=ad_data['group_base_dn'],
+            attr_firstname=LDAP_ATTR['firstname'],
+            attr_lastname=LDAP_ATTR['surname'],
+            attr_login=LDAP_ATTR['login_ad'],
+            server_type=LDAP_SERVER_TYPE['API']['ad'],
+            attr_mail=LDAP_ATTR['mail'],
+            name=gen_string('alpha'),
+            host=ad_data['ldap_hostname'],
+            tls=False,
+            port='389',
+            organization=[module_org],
+            location=[module_loc],
+        ).create()
+        ldap_data = ad_data
+    elif request.param.lower() == 'ipa':
+        # entity create with IPA settings
+        entities.AuthSourceLDAP(
+            onthefly_register=True,
+            account=ipa_data['ldap_user_cn'],
+            account_password=ipa_data['ldap_user_passwd'],
+            base_dn=ipa_data['base_dn'],
+            groups_base=ipa_data['group_base_dn'],
+            attr_firstname=LDAP_ATTR['firstname'],
+            attr_lastname=LDAP_ATTR['surname'],
+            attr_login=LDAP_ATTR['login'],
+            server_type=LDAP_SERVER_TYPE['API']['ipa'],
+            attr_mail=LDAP_ATTR['mail'],
+            name=gen_string('alpha'),
+            host=ipa_data['ldap_hostname'],
+            tls=False,
+            port='389',
+            organization=[module_org],
+            location=[module_loc],
+        ).create()
+        ldap_data = ipa_data
+    else:
```

Comment:
Two non blocking comments, things you might want to adjust in the future as you continue to work on these cases.

1. Many of the fields for AuthSourceLDAP are the same, and these if/else blocks could update a kwargs dictionary containing the common fields with their unique differences, to flatten this out a bit.

2. Forcing parametrization isn't necessary, and forces any test cases that want to use this fixture to parametrize.  Either IPA or AD could be the `else` clause, so that the fixture has a default.  If *all* test cases using this fixture *should* parametrize anyway - then the fixture itself should be parametrized.